include/lsst/afw/geom/SphPoint.h

```diff
@@ -0,0 +1,319 @@
+
+        /**
+         * @brief Angular distance between two points.
+         *
+         * @param other the point to which to measure the separation
+         * @return the length of the shortest arc between the two points. Shall
+         *         not be negative.
+         *
+         * @par Exception Safety:
+         * This method shall not throw exceptions.
+         */
+        Angle separationTo(SphPoint const & other) const noexcept;
+
+        /*
+         * Transformations of points
+         */
+
+        /**
+         * @brief A point defined by rotating this point around an arbitrary axis.
+         *
+         * @param axis the point around which to perform the rotation.
+         * @param amount the angle by which to rotate around @c axis.
+         *               Positive angles represent counterclockwise
+         *               rotations.
+         * @return a new point created by rotating this point
+         *
+         * @throws InvalidParameterError Thrown if @c separation is NaN or infinite.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint rotated(SphPoint const & axis, Angle const & amount) const;
+
+        /**
+         * @brief A point defined by moving along a great circle.
+         *
+         * For any point @c A not at a coordinate pole, and any two angles @c b
+         * and @c delta, <tt>A.bearingTo(A.offset(b, delta))</tt> = @c b and
+         * <tt>A.separationTo(A.offset(b, delta))</tt> = @c delta.
+         *
+         * @param bearing the direction in which to move this point, following
+         *                the conventions described in @ref bearingTo.
+         * @param amount the distance by which to move along the great
+         *               circle defined by @c bearing
+         * @return a new point created by shifting this point
+         *
+         * @throws DomainError Thrown if this point is either coordinate pole.
+         * @throws InvalidParameterError Thrown if either argument is NaN or
+         *                               infinite.
+         * @throws OutOfRangeError Thrown if @c separation is negative.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint offset(Angle const & bearing, Angle const & amount) const;
+
+    private:
+        // For cross-language compatibility, the implementation must be a pair
+        // of floating-point numbers, with no other data. Do not change the
+        // implementation without an RFC.
+        double _longitude;      // radians
+        double _latitude;       // radians
+    };
+
+    /*
+     * Object-level display
+     */
+
+    /**
+     * @brief Prints the value of a point to a stream.
+     *
+     * The point shall be printed as an ordered pair of longitude and (signed)
+     * latitude, in degrees.
```

Comment:
I'd personally go for "SphPoint(10.543250, +32.830583)" to avoid confusion with a cartesian point, but it's your call. They both work. I'm not thrilled about using the default stream precision (and then using the stream operator to code `__str__`, as noted later), but I guess it makes sense.